src/sentry/incidents/charts.py

```diff
@@ -162,8 +163,7 @@ def build_metric_alert_chart(
     end: str | None = None,
     user: Optional["User"] = None,
     size: ChartSize | None = None,
-    query_extra: str | None = None,
-    project_id: int | None = None,
+    subscription: QuerySubscription | None = None,
```

Comment:
rather than passing query_extra here and relying on implementations to construct the query_extra, instead we accept the subscription and construct the query_extra within the chart builder.